### PR TITLE
Add support for parsing local emulator slug

### PR DESCRIPTION
**Pull request by conniey**
# Description

Adds support for parsing local emulator slug.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core-amqp/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 ### Features Added
 
+- Added feature to enable/disable SSL when initially creating connection to support amqp calls on port 5672.
```
Comment (anuchandy) [inline sdk/core/azure-core-amqp/CHANGELOG.md:7]: nit; nit; wondering if using _initially_ make any difference?

_Added feature to enable or disable SSL when creating connection to support AMQP calls on port 5672_

**Review by anuchandy — COMMENTED** [review]
thank you!

**Review by anuchandy — APPROVED** [review]

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 ### Features Added
 
+- Add support for local emulator.
```
[the remark this answers is not in the document]
Comment (conniey) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:7]: They haven't posted a blog post about how to connection, or I would have linked it.  I can add something about the slug though.

File: sdk/core/azure-core-amqp/CHANGELOG.md (hunk 1 above)
Comment (conniey) [reply sdk/core/azure-core-amqp/CHANGELOG.md:7]: Yes. From the [protocol guide](https://learn.microsoft.com/azure/service-bus-messaging/service-bus-amqp-protocol-guide), when you make your initial connection via port 5672, Service Bus will send a request to mandatory upgrade to SSL.  In the case of the emulator though, it won't force you to upgrade when you connect. I hope that helps. :)

Comment (lmolkova) [reply sdk/core/azure-core-amqp/CHANGELOG.md:7]: could you share an example of the connection string here? 
E.g. does it include endpoint (host and port)? also, I assume it does not have to be local - I can run emulator in a docker container and the hostname will be foobar, is this possible?

**Review by lmolkova — APPROVED** [review]

File: sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lmolkova) [reply sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md:7]: could you share an example of the connection string here?
E.g. does it include endpoint (host and port)? also, I assume it does not have to be local - I can run both the app and emulator in a docker container and the hostname will be foobar, is this possible?

File: sdk/core/azure-core-amqp/CHANGELOG.md (hunk 1 above)
Comment (anuchandy) [reply sdk/core/azure-core-amqp/CHANGELOG.md:7]: I see, thanks for the clarification.

**Pull request merged**